### PR TITLE
Support build-artifact-name in module build parameters

### PR DESCRIPTION
**Pull request by tal-sapan**
Before submitting a pull request, Please make sure the following:

## Description

Please explain the changes you made.

Link to design (if applicable)

### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [X] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: internal/artifacts/manifest.go [hunk 1]
```diff
@@ -193,31 +193,52 @@ func getResourcePath(resource *mta.Resource) string {
...
-	// get build results path - defined in build-params property or in
+	// get build results path - defined in the build-result property under build-params or in the module type
 	buildResultPath, _, err := buildops.GetBuildResultsPath(loc, module, defaultBuildResult)
 	if err != nil {
 		return "", err
 	}
 
+	var buildArtifactFileName = ""
+	if module.BuildParams != nil && module.BuildParams[buildArtifactName] != nil {
+		buildArtifactFileName = module.BuildParams[buildArtifactName].(string)
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/manifest.go]: I will. But there are plenty of places in the code where we assume build parameter types (and the type of "path" parameter) without checking if they're correct except the nil check.

File: internal/artifacts/module_arch.go [hunk 2]
```diff
@@ -214,7 +231,7 @@ func copyModuleArchive(ep dir.IModule, modulePath, moduleName string) error {
 		return errors.Wrapf(err, `copying of the "%v" module's archive failed when creating the "%v" folder`, moduleName, moduleTrgZipPath)
 	}
 	moduleTrgZip := filepath.Join(moduleTrgZipPath, "data.zip")
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/module_arch.go]: Done

File: internal/artifacts/manifest.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ import (
 
 const (
 	pathSep        = string(os.PathSeparator)
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/manifest.go]: Done

File: internal/artifacts/manifest.go [hunk 4]
```diff
@@ -193,31 +193,52 @@ func getResourcePath(resource *mta.Resource) string {
 
 func getModulePath(module *mta.Module, targetPathGetter dir.ITargetPath, defaultBuildResult string) (string, error) {
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/manifest.go]: Done

File: internal/artifacts/manifest.go [hunk 5]
```diff
@@ -193,31 +193,52 @@ func getResourcePath(resource *mta.Resource) string {
...
-	return filepath.Base(buildResultPath), nil
+
+	return path, nil
 }
 
-func existsModuleZipInDirectories(module *mta.Module, directories []string) bool {
+func existsFileInDirectories(module *mta.Module, directories []string, filename string) bool {
 	for _, directory := range directories {
-		if _, err := os.Stat(filepath.Join(directory, filepath.ToSlash(module.Name+dataZip))); !os.IsNotExist(err) {
+		if _, err := os.Stat(filepath.Join(directory, filepath.Join(module.Name, filename))); !os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/manifest.go]: Done

File: internal/artifacts/manifest.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/manifest.go]: true, i'll take this as TODO

**Pull request merged**